doc/source/acb_theta.rst

```diff
@@ -0,0 +1,1540 @@
+resulting values will compute the individual Taylor coefficient for each
+derivation tuple that is bounded by `m-1` elementwise. A constant proportion,
+for fixed `g`, of this set consists of all tuples of total order at most
+`m-1`. More precisely, fix `p\in \mathbb{Z}^g`. Then
+
+    .. math ::
+
+        \sum_{n\in \{0,\ldots,m-1\}^g} \zeta^{-p^T n} \theta_{a,b}(z + h_n, \tau)
+        = m^g \sum_{\substack{k\in \mathbb{Z}^g,\ k\geq 0,\\ k = p\ (\text{mod } m)}}
+        a_k\,\varepsilon^{|k|}.
+
+We obtain an upper bound on the tail of this series from the Cauchy integration
+formula: if `|\theta_{a,b}(z,\tau)|\leq c` uniformly on a ball of radius `\rho`
+centered in `z` for `\lVert\cdot\rVert_\infty`, then the sum is `m^g
+(a_p\,\varepsilon^{|p|} + T)` with
+
+    .. math ::
+
+        |T|\leq 2c g\,\frac{\varepsilon^{|p|+m}}{\rho^m}.
+
+Since we divide by `\varepsilon^{|p|}` to get `a_p`, we will add an error of
+`2c g \varepsilon^m/\rho^{m+|p|}` to the result of the discrete Fourier
+transform.
+
+.. function:: void acb_theta_jet_ql_bounds(arb_t c, arb_t rho, acb_srcptr z, const acb_mat_t tau, slong ord)
+
+    Sets *c* and *rho* such that on every ball centered at (a point contained
+    in) *z* of radius *rho*, the functions `|\theta_{a,b}|` for all
+    characteristics `(a,b)` are uniformly bounded by `c`. The choice of *rho*
+    is tuned to get interesting upper bounds on derivatives of `\theta_{a,b}`
+    up to order *ord*.
+
+    We proceed as follows. First, we compute `c_0`, `c_1`, `c_2` such that for
+    any choice of `\rho`, one can take `c = c_0\exp((c_1 + c_2\rho)^2)`
+    above. We can take
+
+        .. math ::
+
+            c_0 = 2^g \prod_{j=0}^{g-1} (1 + 2\gamma_j^{-1}),
+
+        .. math ::
+
+            c_1 = \sqrt{\pi y^T Y^{-1} y},
+
+        .. math ::
+
+            c_2 = \sup_{\lVert x \rVert_\infty\leq 1}
+              \sqrt{\pi x^T \mathrm{Im}(\tau)^{-1} x}.
+
+    One can easily compute an upper bound on `c_2` from the Cholesky
+    decomposition of `\pi \mathrm{Im}(\tau)^{-1}`. We then look for a value of
+    `\rho` that minimizes `\exp((c_1 + c_2\rho)^2)/\rho^{2m-1}` where `m =
+    \mathit{ord}+1`, i.e. we set `\rho` to the positive root of `2c_2\rho
+    (c_1 + c_2\rho) = 2m-1`.
+
+.. function:: void acb_theta_jet_ql_radius(arf_t eps, arf_t err, const arb_t c, const arb_t rho, slong ord, slong g, slong prec)
+
+    Sets *eps* and *err* to be a suitable radius and error bound for computing
+    derivatives up to total order *ord* at precision *prec*, given *c* and
+    *rho* as above.
+
+    We set `varepsilon` such that `(2 g)^{1/m} \varepsilon \leq \rho` and `2 c
+    g \varepsilon^m/\rho^{m+|p|} \leq 2^{-\mathit{prec}}` where `m =
+    \mathit{ord} + 1`. We also set *err* to `2^{-\mathit{prec}}`.
+
+.. function:: void acb_theta_jet_ql_finite_diff(acb_ptr dth, const arf_t eps, const arf_t err, acb_srcptr val, slong ord, slong g, slong prec)
```

Comment:
I have noticed that there is a mix usage of `X_[src]ptr` and `X_t` as inputs into functions. Would it be possible to use either or?